src/test/java/com/thealgorithms/misc/FourSumProblemTest.java

```diff
@@ -0,0 +1,51 @@
+package com.thealgorithms.misc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test cases for The Four Sum Problem.
+ *
+ * @author mouss3abbb
+ */
+
+public class FourSumProblemTest {
+
+    @Test
+    void testFourSumExists() {
+        final var values = new ArrayList<>(List.of(-2, 5, 0, 6));
+        Collections.sort(values);
+        final int target = 9;
+        final var result = FourSumProblem.hashing(values, target).get(0);
+        Collections.sort(result);
+        assertEquals(values, result);
```

Comment:
This test could still pass if `FourSumProblem.hashing(...)` would return more than one solution. Please rewrite the tests checking if the actual result equals the expected result (i.e. single `assertEquals`).

Ideally, please rewrite these tests as parameterized tests. Good example to look at: [`FindMaxRecursionTest`](https://github.com/TheAlgorithms/Java/blob/master/src/test/java/com/thealgorithms/maths/FindMaxRecursionTest.java).

This is a complex problem: please add more tests.